src/dirent.rs

```diff
@@ -0,0 +1,40 @@
+use {Result, Error, Errno};
+use libc::{self, DIR};
+use std::os::unix::io::RawFd;
+use errno;
+
+pub struct Dir {
+    pub handle: *mut DIR,
+}
+
+impl Dir {
+    pub fn read(&mut self) -> Result<Option<&libc::dirent>> {
```

Comment:
borrowing Dir is required here as the dirent reference might get overwritten by subsequent calls.